tests/actor/power_distributing/test_power_distributing.py

```diff
@@ -210,8 +208,9 @@ async def test_power_distributor_one_user(self, mocker: MockerFixture) -> None:
             requests_receiver=requests_channel.new_receiver(),
             results_sender=results_channel.new_sender(),
             component_pool_status_sender=battery_status_channel.new_sender(),
-            wait_for_data_sec=0.1,
         ):
+            await asyncio.sleep(0.1)  # wait for actor to collect data
```

Comment:
Is this PR just moving the issue elsewhere instead? Or is this synchronization only needed for the tests but when actually used one shouldn't care when there is data and when there isn't?

In any case, having tests depend on random `sleep` makes me sad (as I see the flakiness coming to bite us eventually). :cry: :disappointed: 

If we can really really get rid of the need for synchronization, then I completely agree we shouldn't provide any way to synchronize, but otherwise I think having a way to explicitly synchronize that is it not a flaky sleep it is a better approach. In terms of performance, even if it is a hot path, it is just one extra boolean evaluation and branch, which modern CPUs probably won´t even care about because they will do branch prediction so the performance might not even be affected at all after the initialization.

Reply:
As I mentioned in the commit message, these are needed only in tests where we control the PowerDistributor directly.

And they are needed only for those tests that need component data, so that PowerDistributor can return `Success`.  Else it will return `Error("No data")`, but that's not the case we're testing.

It is only a testing problem, not with behaviour in production.

Reply:
OK, I'm still not happy about potential flaky tests, but this is still an improvement over what we had before, so approving :+1: